src/pages/Script/ScriptsComp.tsx

```diff
@@ -0,0 +1,169 @@
+import { useState, useEffect, ReactNode } from 'react'
+import { useHistory } from 'react-router'
+import { Button, Space, Table } from 'antd'
+import { InfoCircleOutlined } from '@ant-design/icons'
+import Pagination from '../../components/Pagination'
+import TransactionItem from '../../components/TransactionItem/index'
+import { fetchTransactionByHash } from '../../service/http/fetcher'
+import i18n from '../../utils/i18n'
+import { ScriptTransactionsPagination, ScriptTransactionsPanel } from './styled'
+import { initTransactionState } from '../../contexts/states/transaction'
+import { TransactionCellDetailModal, TransactionCellInfoPanel } from '../Transaction/TransactionCell/styled'
+import SimpleButton from '../../components/SimpleButton'
+import SimpleModal from '../../components/Modal'
+import TransactionCellScript from '../Transaction/TransactionCellScript'
+import { ScriptInfo, ScriptPageInfo } from './index'
+
+export const ScriptTransactionItem = ({ txHash }: { txHash: string }) => {
+  const [transaction, setTransaction] = useState<State.Transaction>(initTransactionState.transaction)
+
+  useEffect(() => {
+    fetchTransactionByHash(txHash)
+      .then((wrapper: Response.Wrapper<State.Transaction> | null) => {
+        if (wrapper) {
+          const transactionValue = wrapper.attributes
+          if (transactionValue.displayOutputs && transactionValue.displayOutputs.length > 0) {
+            transactionValue.displayOutputs[0].isGenesisOutput = transactionValue.blockNumber === 0
+          }
+          setTransaction(transactionValue)
+        } else {
+          setTransaction(initTransactionState.transaction)
+        }
+      })
+      .catch(() => {
+        setTransaction(initTransactionState.transaction)
+      })
+  }, [txHash])
+
+  return (
+    transaction && (
+      <TransactionItem
+        address=""
+        transaction={transaction}
+        key={txHash}
+        circleCorner={{
+          bottom: false,
+        }}
+      />
+    )
+  )
+}
+
+export const ScriptTransactions = ({
+  scriptInfo,
+  ckbTransaction,
+}: {
+  scriptInfo: ScriptInfo
+  ckbTransaction: ScriptPageInfo<any>
+}) => {
+  const history = useHistory()
+  const totalPages = Math.ceil(ckbTransaction.total / ckbTransaction.size)
+
+  const onChange = (page: number) => {
+    history.replace(`/scripts/${scriptInfo.codeHash}/${scriptInfo.hashType}?page=${page}&size=${ckbTransaction.size}`)
```

Comment:
Why not use `push`, `replace` makes `go back` useless

Reply:
> Why not use `push`, `replace` makes `go back` useless

fixed